src/org/labkey/test/components/ui/grids/CustomizeGridDialog.java

```diff
@@ -0,0 +1,541 @@
+
+        WebElement addIcon = Locator.tagWithClass("div", "view-field__action")
+                .withChild(Locator.tagWithClass("i", "fa-plus"))
+                .findElement(listItem);
+
+        addIcon.click();
+
+        return this;
+    }
+
+    /**
+     * Expand a field or a hierarchy of fields. If a single field is passed in only it will be expanded. If multiple values
+     * are passed in it is assumed to be a path and all fields will be expanded to the last field.
+     *
+     * @param fields The list of fields to expand.
+     * @return This dialog.
+     */
+    public CustomizeGridDialog expandAvailableFields(String... fields)
+    {
+        StringBuilder fieldKey = new StringBuilder();
+
+        Iterator<String> iterator = Arrays.stream(fields).iterator();
+
+        while(iterator.hasNext())
+        {
+            fieldKey.append(iterator.next().replace(" ", ""));
+
+            expandOrCollapseByFieldKey(fieldKey.toString(), true);
+
+            // If this isn't the last item in the collection don't add a "/" to the expected data-fieldkey value.
+            if(iterator.hasNext())
+            {
+                fieldKey.append("/");
+            }
+
+        }
+
+        return this;
+    }
+
+    /**
+     * Collapse a top level field in 'Available Fields' panel.
+     *
+     * @param fieldName Name of the field to collapse.
+     * @return This dialog.
+     */
+    public CustomizeGridDialog collapseAvailableField(String fieldName)
+    {
+        expandOrCollapseByFieldKey(fieldName.replace(" ", ""), false);
+        return this;
+    }
+
+    /**
+     * Private helper function that will expand or collapse a row in the 'Available Fields' panel.
+     *
+     * @param fieldKey The data-fieldkey value of the field to expand.
+     * @param expand True to expand false to collapse.
+     */
+    private void expandOrCollapseByFieldKey(String fieldKey, boolean expand)
+    {
+        WebElement listItem = elementCache().getListItemElementByFieldKey(fieldKey);
+
+        Assert.assertTrue(String.format("Field with data-fieldkey attribute '%s' is not visible in the 'Available Fields' panel.", fieldKey),
+                listItem.isDisplayed());
+
+        String iconClass = expand ? "fa-plus-square" : "fa-minus-square";
+
+        WebElement expandIcon = Locator.tagWithClass("div", "field-expand-icon")
+                .withChild(Locator.tagWithClass("i", iconClass))
+                .findElement(listItem);
+
+        String errorMessage;
+
+        if(expand)
+            errorMessage = String.format("There is no expand icon for field with data-fieldkey attribute '%s' in the 'Available Fields' panel.", fieldKey);
+        else
+            errorMessage = String.format("There is no collapse icon for field with data-fieldkey attribute '%s' in the 'Available Fields' panel.", fieldKey);
+
+        Assert.assertTrue(errorMessage, listItem.isDisplayed());
+
+        expandIcon.click();
+
```

Comment:
This won't work correctly for fields that are already in the desired state. It should no-op for fields that are already in the expected state.
```suggestion
        if (isFieldKeyExpanded(listItem) != expand)
        {
            WebElement expandIcon = Locator.tagWithClass("div", "field-expand-icon").findElement(listItem);
            expandIcon.click();
            
            WebDriverWrapper.waitFor(() -> isFieldKeyExpanded(listItem) == expand,
                    String.format("Field, %s, did not %s.", fieldKey, expand ? "expand" : "collapse"), 5_000);
        }
```

Reply:
I've added a NoOp check at the start that will return if the given row is already in the desired state.